sdk/src/utils/evm.rs

```diff
@@ -0,0 +1,209 @@
+use alloy_transport::BoxTransport;
+use eigensdk::client_avsregistry::writer::AvsRegistryChainWriter;
+use eigensdk::client_elcontracts::reader::ELChainReader;
+use eigensdk::client_elcontracts::writer::{ELChainWriter, Operator};
+use eigensdk::crypto_bls::BlsKeyPair;
+use eigensdk::logging::get_test_logger;
+use lazy_static::lazy_static;
+use rand::random;
+
+lazy_static! {
+    /// 1 day
+    pub static ref SIGNATURE_EXPIRY: U256 = U256::from(86400);
+}
+
+/// Get the provider for a http endpoint
+///
+/// # Returns
+/// - [`RootProvider<BoxTransport>`] - The provider
+///
+pub fn get_provider_http(http_endpoint: &str) -> RootProvider<BoxTransport> {
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .on_http(http_endpoint.parse().unwrap())
+        .root()
+        .clone()
+        .boxed();
+
+    provider
+}
+
+/// Get the provider for a http endpoint with the specified [`Wallet`](EthereumWallet)
+///
+/// # Returns
+/// - [`RootProvider<BoxTransport>`] - The provider
+///
+pub fn get_wallet_provider_http(
+    http_endpoint: &str,
+    wallet: EthereumWallet,
+) -> RootProvider<BoxTransport> {
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(wallet)
+        .on_http(http_endpoint.parse().unwrap())
+        .root()
+        .clone()
+        .boxed();
+
+    provider
+}
+
+/// Get the provider for a websocket endpoint
+///
+/// # Returns
+/// - [`RootProvider<BoxTransport>`] - The provider
+///
+pub async fn get_provider_ws(ws_endpoint: &str) -> RootProvider<BoxTransport> {
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .on_ws(WsConnect::new(ws_endpoint))
+        .await
+        .unwrap()
+        .root()
+        .clone()
+        .boxed();
+
+    provider
+}
+
+/// Get the slasher address from the `DelegationManager` contract
+///
+/// # Returns
+/// - [`Address`] - The slasher address
+///
+/// # Errors
+/// - [`Error::AlloyContract`] - If the call to the contract fails (i.e. the contract doesn't exist at the given address)
+///
+pub async fn get_slasher_address(
+    delegation_manager_addr: Address,
+    http_endpoint: &str,
+) -> Result<Address, Error> {
+    let provider = get_provider_http(http_endpoint);
+    let delegation_manager =
+        eigensdk::utils::binding::DelegationManager::new(delegation_manager_addr, provider);
+    delegation_manager
+        .slasher()
+        .call()
+        .await
+        .map(|a| a._0)
+        .map_err(Error::AlloyContract)
+}
+
+/// Register the account of the given ECDSA pair as an operator on EigenLayer.
+///
+/// # Returns
+/// - [`FixedBytes<32>`] - The transaction hash on success, otherwise an error
+///
+/// # Errors
+/// - [`Error::ElContract`] - If the registration call to the contract itself fails
+/// - [`Error::Keys`] - If the key pair fails to be converted for Alloy
+pub async fn register_as_operator(
```

Comment:
Where is this being used? Should this be the function used on the Eigenlayer runner? Is it just a dangling function for testing?